### ISSUE TITLE
Add negative prefix to cider-refresh for refresh only

### ISSUE THREAD

**Issue by expez** [clojure-emacs/cider#1893]
Most of the time I want `cider-refresh` to call `cider-refresh-before-fn` and `cider-refresh-after-fn` but sometimes I only want the refresh.  I've been using advice to add a negative arity branch to `cider-refresh` and it's pretty nice.

P.S. I've never been in a situation where I wanted to run `before-fn` or `after-fn` only so this won't lead to an explosion in hard to remember prefixes.

**Comment by bbatsov** [comment]
Sounds reasonable to me.

**Comment by dpsutton** [comment]
Well, those functions are clojure code written as strings in elisp. Seems like they won't be too general across all projects. Does it make sense to have the prefix argument invoke the refresh functions and the default refresh behavior not call them? Would you prefer to prefix invoke the refresh functions or prefix escape them, I guess.

**Comment by expez** [comment]
> Well, those functions are clojure code written as strings in elisp.

No, cider just expects a name, .e.g `user/stop` and `user/start`.

> Seems like they won't be too general across all projects.

Not sure why that matters, I want them to not be called when I give negative prefix.  In any event, even though the name is the same, what `user/start` or `user/cider-refresh-before` does in each project can be different.

> Does it make sense to have the prefix argument invoke the refresh functions and the default refresh behavior not call them? Would you prefer to prefix invoke the refresh functions or prefix escape them, I guess.

We already have a prefix and a double prefix behavior, which we can't change without upsetting other people.  So I don't want to change the current prefix options, just add another variant.

**Comment by dpsutton** [comment]
I hadn't thought of their use like that, expez. The examples that I've seen have been defn forms written out. I like the way you use it. Thanks for explaining that.

And I missed that they already had prefix and double prefix. I thought there was only an invocation and the prefix option was blank. I'll get on this.

**Comment by dpsutton** [comment]
So I've got the patch ready, I'm just wondering about a convention:
```

(let ((clear? (member mode '(clear 16)))
        (refresh-all? (member mode '(refresh-all 4)))
        (inhibit-refresh-fns (< mode 0)))
```

This is `cider-refresh` which takes an argument. You can see that it also takes some keywords. I can't find where these are invoked though. I'll add a keyword here but does anyone know where these keywords are used?

**Comment by bbatsov** [comment]
Fixed by #1899.
